lib/Agrammon/DataSource/JSON.pm6

```diff
@@ -0,0 +1,54 @@
+use v6;
+use JSON::Fast;
+
+use Agrammon::Inputs;
+use Agrammon::DataSource::Util;
+
+class Agrammon::DataSource::JSON {
+    method load($simulation-name, $dataset-id, $json-data) {
+        my %input-data = from-json $json-data;
+        my $inputs = Agrammon::Inputs.new(:$simulation-name, :$dataset-id);
+        for %input-data.kv -> $full-tax, $module-data {
+            if $full-tax.index('[') -> $sub-start {
+                my $tax = $full-tax.substr(0, $sub-start);
+                my $sub-end = $full-tax.index(']');
+                my $sub-tax = $full-tax.substr($sub-end + 1);
```

Comment:
Do we want better error handling if the `]` is not found? I think this will probably explode with an exception on the addition.

Reply:
I just realized I can search for `[]` in one go, as the instance name is not inside the parens as it is in CSV. Changed.